packages/office-ui-fabric-react/src/components/Panel/Panel.test.tsx

```diff
@@ -25,6 +25,43 @@ describe('Panel', () => {
     ReactDOM.createPortal.mockClear();
   });
 
+  describe('open', () => {
+    beforeEach(() => {
+      div = document.createElement('div');
+    });
+
+    afterEach(() => {
+      ReactDOM.unmountComponentAtNode(div);
+    });
+
+    it('fires the correct events when opening', done => {
+      let openedCalled = false;
+      let openCalled = false;
+      const setOpenTrue = (): void => {
+        openCalled = true;
+      };
+      const setOpenedTrue = (): void => {
+        openedCalled = true;
+      };
+
+      const panel: PanelBase = ReactDOM.render(<PanelBase isOpen={false} onOpen={setOpenTrue} onOpened={setOpenedTrue} />, div) as any;
+
+      panel.open();
+
+      expect(openCalled).toEqual(true);
+      expect(openedCalled).toEqual(false);
+
+      setTimeout(() => {
+        try {
+          expect(openedCalled).toEqual(true);
+          done();
+        } catch (e) {
+          done(e);
+        }
+      }, 250);
```

Comment:
I am assuming the 250ms here are to wait for the animation of opening the panel to finish. Maybe in this case use `jest.useFakeTimers()` like here:
https://github.com/OfficeDev/office-ui-fabric-react/blob/8ffd8134f6ebbf9ee0785e559e28b146d9de2edf/packages/office-ui-fabric-react/src/components/TextField/TextField.test.tsx#L332

Reply:
@Vitalius1 wouldn't that only apply to setTimeouts and intervals? Does that affect the transition events at all? https://jestjs.io/docs/en/timer-mocks

Reply:
Yes that would work.

Reply:
I just used it over the weekend; it's pretty great, except that it doesn't seem to work well with clearTimeout. (it still seems to execute the callback.) If you hit that, sinon has the same helper and it works better with clearTimeout scenarios. But I think the `jest.useFakeTimers` helper will work here.